log/src/main/java/br/com/codenation/log/service/implementations/LogService.java

```diff
@@ -1,36 +1,120 @@
 import br.com.codenation.log.endpoints.advice.LogNotFoundException;
+import br.com.codenation.log.entity.Usuario;
 import br.com.codenation.log.enums.Ambiente;
 import br.com.codenation.log.enums.Nivel;
 import br.com.codenation.log.enums.Ordenacao;
+import br.com.codenation.log.projection.LogProjection;
+import br.com.codenation.log.repository.LogRepository;
+import br.com.codenation.log.repository.UsuarioRepository;
 import br.com.codenation.log.service.interfaces.LogServiceInterface;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
-import java.util.Collections;
+import java.io.IOException;
+import java.util.Comparator;
 import java.util.List;
-import java.util.Optional;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Service
 @AllArgsConstructor
 public class LogService implements LogServiceInterface {
+
+    private LogRepository repository;
+    private UsuarioRepository usuarioRepository;
+
     @Override
     public LogDTO buscarPorId(Long id) {
-        // TODO: implementar metodo
-        return Optional.of(new LogDTO()).orElseThrow(() -> new LogNotFoundException(id));
+        return repository.findLogById(id)
+                .map(this::mapLogProjectionToLogDTO)
+                .orElseThrow(() -> new LogNotFoundException(id));
     }
 
     @Override
-    public List<LogDTO> listarComFiltros(Ambiente ambiente, Nivel nivel, String descricao, String origem, Ordenacao ordenacao) {
-        // TODO: implementar metodo
-        return Collections.emptyList();
+    public List<LogDTO> listarComFiltros(Ambiente ambiente,
+                                         Nivel nivel,
+                                         String descricao,
+                                         String origem,
+                                         Ordenacao ordenacao) {
+
+        if (Objects.nonNull(nivel))
+            return filtraPorNivel(ambiente, nivel, ordenacao);
+
+        if (Objects.nonNull(descricao))
+            return filtraPorDescricao(ambiente, descricao, ordenacao);
+
+        if (Objects.nonNull(origem))
+            return filtraPorOrigem(ambiente, origem, ordenacao);
+
+        return filtraPorAmbiente(ambiente);
+    }
+
+    private List<LogDTO> filtraPorAmbiente(Ambiente ambiente) {
+        return mapLogProjectionToLogDTO(repository.findAllByAmbiente(ambiente.toString()));
+    }
+
+    private List<LogDTO> filtraPorNivel(Ambiente ambiente, Nivel nivel, Ordenacao ordenacao) {
+        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndNivel(ambiente.toString(), nivel.toString()));
+        return ordenaLista(logs, ordenacao);
+    }
+
+    private List<LogDTO> filtraPorDescricao(Ambiente ambiente, String descricao, Ordenacao ordenacao) {
+        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndDescricao(ambiente.toString(), descricao));
```

Comment:
```suggestion
        List<LogDTO> logs = mapLogProjectionToLogDTO(repository.findAllByAmbienteAndDescricao(ambiente.name(), descricao));
```
Ao invés de usar o `toString()` nesse caso é melhor usar o `name`.